### ISSUE TITLE
GraphQL query support

### ISSUE THREAD

**Issue by Flushot** [luckymarmot/Paw-Issues#57]
It would be nice if I could easily query GraphQL API endpoints in Paw like I can do with [Insomnia](https://insomnia.rest/blog/introducing-graphql/) and [GraphiQL](https://github.com/graphql/graphiql). I'd much rather use Paw than have to switch REST clients for this specific task.

**Comment by dlo** [comment]
Integrating with the GitHub v4 API right now and having to manipulate the query manually is a little bit of work. Throwing my hat in the ring to request this too. :)

**Comment by jamesggraf** [comment]
I would love to see this too. I hate having to switch to GraphiQL

**Comment by eegilbert** [comment]
Another Me Too.

**Comment by alec-c4** [comment]
+1

**Comment by Leandros** [comment]
+1

**Comment by shawnbayer** [comment]
+1

**Comment by kimochg** [comment]
+1

**Comment by jordanhamill** [comment]
+1

**Comment by StummeJ** [comment]
+1

**Comment by pistelak** [comment]
+1

**Comment by Vybo** [comment]
+1

**Comment by sjavora** [comment]
+1

**Comment by RadimLanger** [comment]
+1

**Comment by infotrinerdis** [comment]
+1

**Comment by mullerfrederik** [comment]
+1

**Comment by martiis** [comment]
+1

**Comment by KevinHurts** [comment]
+1

**Comment by najeeb-rifaat** [comment]
+1

**Comment by travy** [comment]
+1

**Comment by markgaensicke** [comment]
+1

**Comment by sthtodo** [comment]
+1

**Comment by chrisb** [comment]
+1

**Comment by sibljon** [comment]
+1

**Comment by DamianLion** [comment]
+1

**Comment by SeriousSem** [comment]
+1

**Comment by sheharyarn** [comment]
FYI: Until Paw implements GraphQL, you should check out [Insomnia](https://insomnia.rest/). They added [GraphQL Support](https://insomnia.rest/graphql/) a while back, and it works pretty well. Much better than _GraphiQL_, at least.

**Comment by joelgetaction** [comment]
Any word?

**Comment by cip22** [comment]
+1

**Comment by larryonoff** [comment]
+1

**Comment by anonrig** [comment]
Any news from the Paw team about this?

**Comment by maxromanovsky** [comment]
Everybody hates receiving email notification with `+1` from GH issue, but here you go: `+1` :)

**Comment by techprimate-phil** [comment]
+1

**Comment by zhuzhichao** [comment]
[https://github.com/imolorhe/altair]   is a better tool for GraphQL now.

**Comment by emil-nasso** [comment]
I find that insomnia is pretty decent for this. I need a graphql-capable client so I have started migrating from paw to insomnia for my REST-stuff too.

Can recommend insomnia. It's pretty nice. I like to have a native desktop app for my http-client, not an browser extension.

**Comment by techprimate-phil** [comment]
Insomnia is pretty good piece of software, but I am missing the option `Response Body Dynamic Value` you can use in Paw

**Comment by stedwick** [comment]
Switched to insomnia for now

**Comment by samgranieri** [comment]
I love working with Paw, and I’d like to consolidate my GQL work in the same app I use for REST api development.

**Comment by bukhalo** [comment]
one year has passed

**Comment by joelgetaction** [comment]
Try tweeting them on Twitter since they don’t seem to respond here:   
  

  
http://twitter.com/luckymarmot
  

  
  

  

  
  
>

**Comment by joshhopkins** [comment]
Very much needed. There a bunch of decent electron apps out there, but none provide the stability and small footprint of Paw.

**Comment by sayduck-daniel** [comment]
It would be great to handle subscriptions as well. GraphiQL Electron app is nice but does not support it sadly.

**Comment by imolorhe** [comment]
@sayduck-daniel @joshhopkins  [Altair](https://altair.sirmuel.design/) does.

**Comment by joshhopkins** [comment]
@imolorhe Altair is nice, but I'd prefer to use the same HTTP client for both gql and rest. Plus my machine doesn't need yet another node process running electron.

**Comment by imolorhe** [comment]
@joshhopkins Yeah true, I understand. You could also use the browser extensions Altair provides though.

**Comment by wmai** [comment]
+ 1

**Comment by sayduck-daniel** [comment]
> @sayduck-daniel @joshhopkins [Altair](https://altair.sirmuel.design/) does.

Yes but let's be honest, Altair does 1% of what Paw does, Paw is way more stable and future proof than Altair.

I already pay for Paw, so if it could be compatible with all kinds of APIs, I wouldn't use other tools for that.

**Comment by eivindml** [comment]
Would really like to see this as well.

**Comment by dcypherthis** [comment]
another upvote 👍

**Comment by bad884** [comment]
Please please add this... and soonish

**Comment by Tyncture** [comment]
Highly interested in this too.

**Comment by ruslaniv** [comment]
I would like this feature too! But here's [**2 year old tweet**](https://twitter.com/luckymarmot/status/895222031146520576) requesting this very feature and it is still not implemented. So I guess it's about time to switch clients.

**Comment by mikeyaa** [comment]
Any update on this feature? Getting embarrassing that PAW doesn't support GraphQL

**Comment by jakubczaplicki** [comment]
Meanwhile.. Postman has very nice support for GraphQL..

**Comment by StummeJ** [comment]
So does the opensource Insomnia rest client...

**Comment by hugo082** [comment]
+1

**Comment by williamsthing** [comment]
+1

**Comment by Jax0rz** [comment]
+1

**Comment by zhuzhichao** [comment]
+1

**Comment by gordonk** [comment]
Well past time we had this. +1

**Comment by calbertts** [comment]
+1

**Comment by glaurent** [comment]
+1

**Comment by davizucon** [comment]
+1

**Comment by austinmills** [comment]
+1

**Comment by d8vjork** [comment]
Another one waiting for this, I won't pay a license unless I've this (I already use Postman which have a decent integration with almost all the API kinds)

**Comment by hristaki** [comment]
Do we have an official comment for this feature request from the developers?
The issue is created on 16th Jan 2018

**Comment by d8vjork** [comment]
@hristaki 

https://twitter.com/luckymarmot/status/1174055485378650112

**Comment by joshhopkins** [comment]
Good to see LuckyMarmot are still alive. How difficult would it be to update this issue.

**Comment by d8vjork** [comment]
Seems like nobody is checking their Github though (better support on Twitter)

**Comment by joshhopkins** [comment]
@d8vjork seems that way. Why have this public repo then 🤔

**Comment by CharlieBreval** [comment]
+1 we absolutly need it

**Comment by davizucon** [comment]
+1

**Comment by fordnox** [comment]
+1

**Comment by seanhealy** [comment]
Looking like it might be close here… Can't wait!

https://twitter.com/luckymarmot/status/1192433987177918465

**Comment by zhuzhichao** [comment]
![image](https://user-images.githubusercontent.com/3017759/70715985-62192800-1d26-11ea-9ebd-c03d9756d01e.png)

Today is 12.12, it means Paw for GraphQL is coming ?

**Comment by oshalygin** [comment]
Its `12.19`

I reached out to beta test but there has been no response.  My guess is maybe sometime next year instead?

**Comment by ole3021** [comment]
it `12.25`  🌲

Merry Christmas and please release Graphql with Paw

**Comment by CharlieBreval** [comment]
It s `12.31` 🙄 
Time for a surprise

**Comment by eivindml** [comment]
You all are very entitled. It will be released when it's ready. No need to spam this issue with the same posts. 

🙏🏼

**Comment by zakiaziz** [comment]
@eivindml Some of us paid for this software under the impression that this'd be released around a certain time frame. Entitled are the ones that didn't pay and are asking for updates.

**Comment by ole3021** [comment]
Promise should be promise, if it's delayed for some reason, I think there should have a reason and new release date announced, otherwise how I can trust what you say and the software as well. we "spam" this issue cause we have nowhere else can say it out and get the team's attention.  Sorry for the spam.

**Comment by sayduck-daniel** [comment]
Is it ready?

**Comment by gordonk** [comment]
OMG I'm dying here!

**Comment by ole3021** [comment]
Switched to Postman, sorry paw.

**Comment by wiesson** [comment]
Any updates? :)

**Comment by ukstv** [comment]
Well, it was prematurely for me to buy Paw, having no GraphQL support landed.

**Comment by bad884** [comment]
This needs to be done.

**Comment by eX-innovation** [comment]
Any updates? Can't wait any longer

**Comment by eivindml-solvr** [comment]
Pretty sure they're not using this repo. The last issues where closed by the issue authors themselves, so no official activity here since 2018 or something. 

<img width="1007" alt="Skjermbilde 2020-02-18 kl  15 41 01" src="https://user-images.githubusercontent.com/55379708/74746087-1ecac080-5265-11ea-9a82-3b99eef000a8.png">

**Comment by zakiaziz** [comment]
Hey guys, I don't think we'll be seeing anything for another 3 or so months:


![Screen Shot 2020-03-05 at 12 36 47 PM](https://user-images.githubusercontent.com/940281/76009104-a871c380-5ede-11ea-9b34-b9a831af392c.png)

**Comment by oshalygin** [comment]
Anyone have a nice workaround or extension that can get us most of the way there?  I tried importing a HAR and while it "works", wondering if folks are coming up with better solutions. 

Please do not recommend Postman, thats a non-starter.

**Comment by andrewbellucci** [comment]
I don't mind the wait, it's been a long time in the making, but it will be worth it in the end!

**Comment by jdecarlo** [comment]
you can use Paw to query graphql just using raw _Test_ or _JSON_ as long as you don't put newlines in the query value.  

For example,

{
   "query" : "query { users { name address { street city state } }}"
}

It's a pain and not as pretty as Postman's implementation, but it works.  

Paw - need to get on this!  Postman and other competitors are out pacing you!

**Comment by natesilva** [comment]
I’d really like to see something more like Insomnia than Postman. I can see docs, errors are highlighted, schemas don’t need to be imported.

I’m glad to hear they’re working on this. Nothing else is as good as Paw.

**Comment by torcelly** [comment]
Any news? I need GraphQL support guys. Thanks

**Comment by donbeave** [comment]
Agree, Insomnia implementation is much better.

**Comment by tylucaskelley** [comment]
Any update on this?

**Comment by zakiaziz** [comment]
hold on boys, it's gonna be a while longer
<img width="1037" alt="Screen Shot 2020-06-22 at 7 51 08 PM" src="https://user-images.githubusercontent.com/940281/85345907-cf6df880-b4c1-11ea-8a18-69848b38e945.png">

**Comment by zakiaziz** [comment]
When asked about a rough timeline:
<img width="512" alt="Screen Shot 2020-06-25 at 10 53 30 AM" src="https://user-images.githubusercontent.com/940281/85743542-1cc0b480-b6d2-11ea-8702-82b79360f623.png">

**Comment by DamianLion** [comment]
I mean at least there are responsive

**Comment by bukhalo** [comment]
> We're about to release an exciting update to Paw. Among other features, it will bring support for macOS 11 Big Sur and GraphQL APIs.

http://blog.paw.cloud/stay-updated-beta/

**Comment by zakiaziz** [comment]
Will this update also be available in setapp?

**Comment by natesilva** [comment]
This is great news! It’s been years since I paid for my license (besides SetApp, but I can’t imagine you make much from that). Will happily pay for an upgrade if the GraphQL functionality is good.

**Comment by amomchilov** [comment]
HYPE

<img width="838" alt="image" src="https://user-images.githubusercontent.com/5703449/94486573-9d99bb80-01ad-11eb-8417-51d3a14bd2fa.png">

**Comment by pbojinov** [comment]
Can anyone share the Paw 3.2 Beta link referenced above?

**Comment by sgleadow** [comment]
@pbojinov I think you can get it from this page? https://blog.paw.cloud/paw-2-1-public-beta/

So far I'm liking it.

**Comment by nboukeffa** [comment]
I think this issue can be closed since Paw 3.2 has been released for everyone 🎉🎊🥳

https://blog.paw.cloud/paw-3-2-graphql-macos-big-sur/

**Comment by mittalyashu** [comment]
When will Paw 3.2 be released on Setapp?

**Comment by mittalyashu** [comment]
Paw 3.2 is still not available via Setapp, showing older version.

![CleanShot 2020-10-21 at 10 34 50@2x](https://user-images.githubusercontent.com/29014463/96675686-12b17880-1389-11eb-88c8-2ec3702c9405.png)

**Comment by wiesson** [comment]
> Paw 3.2 is still not available via Setapp, showing older version.
> 
> ![CleanShot 2020-10-21 at 10 34 50@2x](https://user-images.githubusercontent.com/29014463/96675686-12b17880-1389-11eb-88c8-2ec3702c9405.png)

Do you mind opening a new ticket for this? This issue is about graphql not Setapp issues! 🙏🏻

**Comment by zakiaziz** [comment]
Should be coming to setapp sometime this week according to Piotr:

> <img width="697" alt="Screen Shot 2020-10-21 at 11 38 38 AM" src="https://user-images.githubusercontent.com/940281/96750788-fcf48100-1391-11eb-8fcf-6ca81a99582e.png">

**Comment by sscotth** [comment]
Enjoying it so far. Some low hanging fruit features would be a commenting shortcut `⌘ + /` and format/prettify.

**Comment by natesilva** [comment]
I believe someone also mentioned there should be a prettify/format feature. I too had a hard time finding that feature. It’s located under Edit > Pretty-print Text (shortcut <kbd>Ctrl</kbd>-<kbd>I</kbd>).

**Comment by sscotth** [comment]
Yep, it's a bit hidden, would be nice if there was a quick access button.

**Comment by sharukh967** [comment]
Is there a way to populate all the fields of an object?
Altair GraphQL client and Insomnia have this feature.
